pkg/apiserver/resthandler.go

```diff
@@ -171,6 +172,85 @@ func CreateResource(r RESTCreater, ctxFn ContextFunc, namer ScopeNamer, codec ru
 	}
 }
 
+// PatchResource returns a function that will handle a resource patch
+func PatchResource(r RESTPatcher, ctxFn ContextFunc, namer ScopeNamer, codec runtime.Codec, resource string, admit admission.Interface) restful.RouteFunction {
+	return func(req *restful.Request, res *restful.Response) {
+		w := res.ResponseWriter
+
+		// TODO: we either want to remove timeout or document it (if we document, move timeout out of this function and declare it in api_installer)
+		timeout := parseTimeout(req.Request.URL.Query().Get("timeout"))
+
+		namespace, name, err := namer.Name(req)
+		if err != nil {
+			notFound(w, req.Request)
+			return
+		}
+		ctx := ctxFn(req)
+		ctx = api.WithNamespace(ctx, namespace)
+
+		original, err := r.Get(ctx, name)
+		if err != nil {
+			errorJSON(err, codec, w)
+			return
+		}
+
+		originalObjJs, err := codec.Encode(original)
+		if err != nil {
+			errorJSON(err, codec, w)
+			return
+		}
+		patchJs, err := readBody(req.Request)
+		if err != nil {
+			errorJSON(err, codec, w)
+			return
+		}
+		patchedObjJs, err := merge.MergeJSON(originalObjJs, patchJs)
+		if err != nil {
+			errorJSON(err, codec, w)
+			return
+		}
+
+		obj := r.New()
+		if err := codec.DecodeInto(patchedObjJs, obj); err != nil {
+			errorJSON(err, codec, w)
+			return
+		}
+		if err := checkName(obj, name, namespace, namer); err != nil {
+			errorJSON(err, codec, w)
+			return
+		}
+
+		// PATCH requires same permission as UPDATE
+		err = admit.Admit(admission.NewAttributesRecord(obj, namespace, resource, "UPDATE"))
+		if err != nil {
+			errorJSON(err, codec, w)
+			return
+		}
+
+		wasCreated := false
```

Comment:
No, and the previous get would fail


> On Feb 23, 2015, at 4:21 PM, Daniel Smith <notifications@github.com> wrote:
> 
> In pkg/apiserver/resthandler.go:
> 
> > +			errorJSON(err, codec, w)
> > +			return
> > +		}
> > +		if err := checkName(obj, name, namespace, namer); err != nil {
> > +			errorJSON(err, codec, w)
> > +			return
> > +		}
> > +
> > +		// PATCH requires same permission as UPDATE
> > +		err = admit.Admit(admission.NewAttributesRecord(obj, namespace, resource, "UPDATE"))
> > +		if err != nil {
> > +			errorJSON(err, codec, w)
> > +			return
> > +		}
> > +
> > +		wasCreated := false
> Is it normal to allow creating something with a PATCH request?
> 
> —
> Reply to this email directly or view it on GitHub.
>